File: Convert_an_Array_into_a_2D_Array.cpp
// 2610. Convert an Array Into a 2D Array With Conditions
// Difficulty: Medium
// Problem: https://leetcode.com/problems/convert-an-array-into-a-2d-array-with-conditions/description

class Solution {
public:
    vector<vector<int>> findMatrix(vector<int>& nums) {
        vector<vector<int>> ans;
        map <int, int> counter;

        for (int num : nums)
        {
            if (counter.find(num) == counter.end()){
                counter[num] = 1;
            }
            else{
                counter[num]++;
            }

            if (ans.size() < counter[num])
            {
                ans.push_back({num});
            }
            else{
                ans[counter[num]-1].push_back(num);
            }
        }
        return ans;
    }
};

File: Convert_an_Array_into_a_2D_Array.py
# TItle: 2610. Convert an Array Into a 2D Array With Conditions
# Difficulty: Medium
# Problem: https://leetcode.com/problems/convert-an-array-into-a-2d-array-with-conditions/description

class Solution:
    def findMatrix(self, nums: List[int]) -> List[List[int]]:
        ans, counter = [], {}

        for num in nums:
            if num not in counter:
                counter[num] = 0
            counter[num] += 1

            if counter[num] > len(ans):
                ans.append([])
            
            ans[counter[num]-1].append(num)
        return ans
